src/commands/init.ts

```diff
@@ -82,160 +112,215 @@ export abstract class InitCommand extends Command {
+      environment: { architect_property: 'environment', func: (environment: any) => { return { base: environment }; } },
+      command: { architect_property: 'command', func: (command: any) => { return { base: command }; } },
+      entrypoint: { architect_property: 'entrypoint', func: (entrypoint: any) => { return { base: entrypoint }; } },
+      image: { architect_property: 'image', func: (image: string) => { return { base: image }; } },
+      build: { architect_property: 'build', func: this.convertBuild },
+      ports: { architect_property: 'interfaces', func: this.convertPorts },
+      volumes: { architect_property: 'volumes', func: this.convertVolumes },
+      depends_on: { architect_property: 'depends_on', func: this.convertDependsOn },
+      external_links: { architect_property: 'depends_on', func: this.convertDependsOn },
+    };
+
     const architect_component: Partial<ComponentSpec> = {};
     architect_component.name = `${flags.account || account.name}/${flags.name || answers.name}`;
     architect_component.services = {};
-    for (const [service_name, service] of Object.entries(docker_compose.services || {})) {
+    for (const [service_name, service_data] of Object.entries(docker_compose.services || {})) {
       const architect_service: Partial<ServiceSpec> = {};
-      architect_service.description = `${service_name} converted to an Architect service with "architect init"`;
-      architect_service.environment = service.environment;
-      architect_service.command = service.command;
-      architect_service.entrypoint = service.entrypoint;
-
-      if (service.image) {
-        architect_service.image = service.image;
-      } else if (service.build) {
-        architect_service.build = {};
-        if (typeof service.build === 'string') {
-          architect_service.build.context = service.build;
-        } else {
-          if (Array.isArray(service.build.args)) {
-            architect_service.build.args = {};
-            for (const arg of service.build.args) {
-              const [key, value] = arg.split('=');
-              if (key && value) {
-                architect_service.build.args[key] = value;
-              } else {
-                this.warn(chalk.yellow(`Could not convert environment variable ${arg} for service ${service_name}`));
-              }
+      for (const [property_name, property_data] of Object.entries(service_data || {})) {
+        if (compose_property_converters[property_name]) {
+          const architect_property_name = compose_property_converters[property_name].architect_property;
+          const converted_props: ComposeConversion = compose_property_converters[property_name].func(property_data, docker_compose, architect_service);
+          if (converted_props.local) {
+            if (!(architect_service as any)[this.local_config_placeholder]) { // TODO: how do we add interpolated fields more cleanly to the ServiceSpec?
```

Comment:
Could you give an example?

Reply:
Since of course conditionals aren't keys in our service spec, I had to convert to `any` to be able to accommodate adding these local properties. Felt kind of weird but if it doesn't come up anywhere else it doesn't seem like that big of a deal either